### DISCUSSION TITLE
Alternative Switch Statement Syntax

### DISCUSSION THREAD

**Discussion by benjaminiserman** [dotnet/csharplang#5534] [Language Ideas]
The current syntax for switch statements strikes me as a strange carryover from C. Why are switch statements the only construct in C# that uses `:` and `break` to control code flow? This also adds strange behavior with which variables are in scope, leading to some developers just adding curly braces to cases anyways. Considering the fact that C# doesn't allow case fall-through, the C syntax here seems dated to me. I'm curious to hear the community's thoughts on adding an additional syntax for switch statements. Example:
```cs
int? x;

switch (x)
{
	case < 0:
		Console.WriteLine("Negative.");
		break;
	case > 0:
		Console.WriteLine("Positive.");
		break;
	case 0:
		Console.WriteLine("Zero.");
		break;
	default:
		Console.WriteLine("Null");
		break;
}
```
might become:
```cs
int? x;

switch (x)
{
	case (< 0)
	{
		Console.WriteLine("Negative.");
	}
	case (> 0)
	{
		Console.WriteLine("Positive.");
	}
	case (0)
	{
		Console.WriteLine("Zero.");
	}
	default
	{
		Console.WriteLine("Null");
	}
}
```
It seems to me that this would add little complexity to the language, given its similarity to other language constructs. As far as I'm aware this wouldn't cause any breaking changes either, and it feels unambiguous with current switch statements. Does anyone have any thoughts on this?

**Comment by 333fred** [comment]
See https://github.com/dotnet/csharplang/issues/3038.

**Comment by CyrusNajmabadi** [comment]
> Does anyone have any thoughts on this?

Yes.  In general we would only want to add new syntax for a feature if there was *super high* value in doing so.  There are real costs (in implementation, learning, and complexity, etc.) to adding new features.  So we will only do these things if the net benefit is large enough to make it worthwhile.

Having something being a holdover from C is fine.  It's known, has been consistent for 20 years (for our language).  It's 'dated' just as much as so much of this language is (being 20 years old now).  So this very much falls into the "if it ain't broke, don't fix it category".  

Note: this does not mean this wouldn't be done.  Just that it requires a lot of benefits beyond just looking 'more modern'.
